source/vir/src/ast.rs

```diff
@@ -89,9 +89,30 @@ pub enum IntRange {
     ISize,
 }
 
+/// Type information relevant to Rust but generally not relevant to the SMT encoding.
+/// This information is relevant for resolving traits.
+#[derive(Debug, Serialize, Deserialize, Hash, ToDebugSNode, Clone, Copy, PartialEq, Eq)]
+pub enum TypDecoration {
+    /// &T
+    Ref,
+    /// &mut T
+    MutRef,
```

Comment:
Why is MutRef here?

(I guess that this does nothing at the moment, due to all the current restrictions on `&mut`. But either why I don't think it makes sense to consider it isomorphic to `T` and `&T`)

Reply:
Rust lets trait implementations specialize on `&` and `&mut`:

```
trait T { fn f(&self) -> u64; }
struct S {}
impl T for S { fn f(&self) -> u64 { 3 } }
impl T for &S { fn f(&self) -> u64 { 4 } }
impl T for &mut S { fn f(&self) -> u64 { 5 } }
```

Reply:
Is `impl T for &mut S { fn f(&self) -> u64 { 5 } }` actually supported right now? Given our limited support for `&mut` types, it's hard to see how that would work.

What I'm really asking is ... I thought type decorations were for "trivial" type constructors, like `&` or `Box<_>` which would otherwise be dropped entirely from the encoding, but that can't be the case for `&mut`.